plugins/catalog-backend/src/database/DatabaseManager.ts

```diff
@@ -18,17 +18,36 @@ import Knex from 'knex';
 import path from 'path';
 import { Logger } from 'winston';
 import { CommonDatabase } from './CommonDatabase';
-import type { Database } from './types';
+import { Database } from './types';
 
 export class DatabaseManager {
   public static async createDatabase(
-    knex: Knex,
+    database: Knex,
     logger: Logger,
   ): Promise<Database> {
-    await knex.migrate.latest({
+    await database.migrate.latest({
       directory: path.resolve(__dirname, 'migrations'),
       loadExtensions: ['.js'],
     });
-    return new CommonDatabase(knex, logger);
+    return new CommonDatabase(database, logger);
+  }
+
+  public static async createInMemoryDatabase(
+    logger: Logger,
+  ): Promise<Database> {
+    const database = Knex({
+      client: 'sqlite3',
+      connection: ':memory:',
+      useNullAsDefault: true,
+    });
+    database.client.pool.on('createSuccess', (_eventId: any, resource: any) => {
+      resource.run('PRAGMA foreign_keys = ON', () => {});
+    });
+
+    await database.migrate.latest({
```

Comment:
at this point you should probably do `return DatabaseManager.createDatabase(database, logger);` to not repeat the code unnecessarily